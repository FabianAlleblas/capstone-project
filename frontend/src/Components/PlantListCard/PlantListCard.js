import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import defaultPlantImage from '../../assets/plant-images/default_plant.jpg'
import PlantListBar from '../IndicatorBars/PlantListBar'

export default function PlantListCard({ plant, onClick }) {
  PlantListCard.propTypes = {
    plant: PropTypes.object.isRequired,
    onClick: PropTypes.func.isRequired,
  }

  return (
    <CardWrapper onClick={onClick}>
      <ImgContainer src={plant.image ?? defaultPlantImage} />
      <PlantName>{plant.name}</PlantName>
      <PlantListBar daysLeft={plant.daysLeft} weeksLeft={plant.weeksLeft} />
    </CardWrapper>
  )
}

const CardWrapper = styled.div`
  display: grid;
  place-items: center;
`

const ImgContainer = styled.div`
  background-image: url(${(props) => props.src});
  background-position: center;
  background-size: cover, contain;
  border-radius: 50%;
  height: 100px;
  margin-bottom: 10px;
  width: 100px;
`

const PlantName = styled.h5`
  color: var(--primary-plant-font-color);
  font-size: 1.25rem;
  font-weight: 400;
`
