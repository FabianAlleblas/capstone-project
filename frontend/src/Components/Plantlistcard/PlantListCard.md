```js
<PlantListCard
  name={'Jack'}
  species={'Monstera deliciosa'}
  onClick={() => {}}
/>
```
