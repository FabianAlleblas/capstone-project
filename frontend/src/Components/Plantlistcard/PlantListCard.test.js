import { render } from '@testing-library/react'
import 'jest-styled-components'
import PlantListCard from './PlantListCard'

describe('PlantCard', () => {
  it('renders correctly', () => {
    const { container, getByText } = render(
      <PlantListCard plant={{ name: 'Jack', daysLeft: 4, weeksLeft: 2 }} />
    )

    expect(container.firstChild).toMatchSnapshot()
    expect(getByText(/Jack/i)).toBeInTheDocument()
  })
})
