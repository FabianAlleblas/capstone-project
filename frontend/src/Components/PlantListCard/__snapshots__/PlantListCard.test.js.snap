// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlantCard renders correctly 1`] = `
.c3 {
  display: grid;
  gap: 6px 6px;
  grid-template-columns: 15px auto;
  padding: 10px 0 0;
  place-items: center;
  width: 100%;
}

.c4 {
  background-color: var(--bar-bg-color);
  border-radius: 1px;
  height: 2px;
  position: relative;
  width: 100%;
}

.c5 {
  border-radius: 3px;
  height: 6px;
  min-width: 5px;
  position: absolute;
  top: -2px;
}

.c6 {
  background-color: var(--bar-water-color);
  width: 40%;
}

.c7 {
  background-color: var(--bar-fertilizer-color);
  width: 50%;
}

.c0 {
  display: grid;
  place-items: center;
}

.c1 {
  background-image: url(default_plant.jpg);
  background-position: center;
  background-size: cover,contain;
  border-radius: 50%;
  height: 100px;
  margin-bottom: 10px;
  width: 100px;
}

.c2 {
  color: var(--primary-plant-font-color);
  font-size: 1.25rem;
  font-weight: 400;
}

<div
  class="c0"
>
  <div
    class="c1"
    src="default_plant.jpg"
  />
  <h5
    class="c2"
  >
    Jack
  </h5>
  <div
    class="c3"
  >
    <svg>
      watericon.svg
    </svg>
    <div
      class="c4"
    >
      <div
        class="c5 c6"
      />
    </div>
    <svg>
      fertilizericon.svg
    </svg>
    <div
      class="c4"
    >
      <div
        class="c5 c7"
      />
    </div>
  </div>
</div>
`;
